File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/armor/effects/UnbreakableArmorEffect.java
package com.pokemod.pokemod.world.item.armor.effects;

import com.pokemod.pokemod.world.item.armor.PokeModArmorItem;
import com.pokemod.pokemod.world.item.armor.ArmorEffect;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;

public record UnbreakableArmorEffect() implements ArmorEffect {
    @Override
    public void onArmorTick(ItemStack itemStack, Level world, Player player, PokeModArmorItem pokeModArmorItem) {
        if (world.isClientSide) return;
        if (itemStack.getOrCreateTag().getBoolean("Unbreakable")) return;
        itemStack.getOrCreateTag().putBoolean("Unbreakable", true);
    }

    @Override
    public void inventoryTick(ItemStack itemStack, Level world, Entity entity, int slotId, boolean isSelected, PokeModArmorItem pokeModArmorItem) {

    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/item/armor/effects/UnbreakableArmorEffect.kt
package generations.gg.generations.core.generationscore.common.world.item.armor.effects

import generations.gg.generations.core.generationscore.common.world.item.armor.ArmorTickEffect
import generations.gg.generations.core.generationscore.common.world.item.armor.GenerationsArmorItem
import net.minecraft.core.component.DataComponents
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.component.Unbreakable
import net.minecraft.world.level.Level

class UnbreakableArmorEffect : ArmorTickEffect {
    override fun inventoryTick(
        itemStack: ItemStack,
        world: Level,
        entity: LivingEntity,
        slotId: Int,
        isSelected: Boolean,
        generationsArmorItem: GenerationsArmorItem
    ) {
        if (world.isClientSide()) return
        if (itemStack.has(DataComponents.UNBREAKABLE)) return
        itemStack.set(DataComponents.UNBREAKABLE, Unbreakable(false))
    }
}
